### PR TITLE
Fix L0_build_variants

### PR DESCRIPTION
**Pull request by CoderHam**
- Fix No GPU server build
- Fix build without stats collection

@GuanLuo can you have a look at the onnx change I made? I appears the string cuda copy check was always false.

### CODE REVIEW

**Comment by tanmayv25** [comment]
@CoderHam  Did you check L0_build_variants with these changes?

**Comment by CoderHam** [comment]
> @CoderHam Did you check L0_build_variants with these changes?

Yes, all tests except _Build without stats collection_ pass. I have a fix for that on the gitlab side by turning TRITON_ENABLE_METRICS=OFF as it fails only because of this check:
https://github.com/NVIDIA/triton-inference-server/blob/master/build/CMakeLists.txt#L77-L79

File: src/backends/onnx/onnx_backend.cc [hunk 1]
```diff
@@ -854,13 +853,6 @@ OnnxBackend::Context::SetInputTensors(
         buffer_offset += expected_byte_sizes[ridx];
       }
 
-#ifdef TRITON_ENABLE_GPU
-      // Synchronize to ensure the buffer is ready to be modified
-      if (string_cuda_copy) {
-        cudaStreamSynchronize(stream_);
-      }
-#endif  // TRITON_ENABLE_GPU
```
Comment (GuanLuo) [inline src/backends/onnx/onnx_backend.cc:862]: Why dropping this?

File: src/core/cuda_utils.h [hunk 2]
```diff
@@ -47,7 +47,8 @@ namespace nvidia { namespace inferenceserver {
 
 #ifndef TRITON_ENABLE_GPU
 using cudaStream_t = void*;
-#endif  // TRITON_ENABLE_GPU
+using cudaEvent_t = void*;
```
Comment (GuanLuo) [inline src/core/cuda_utils.h:50]: I don't think `cudaEvent_t` is mentioned in cuda_utils?

File: src/backends/onnx/onnx_backend.cc (hunk 1 above)
Comment (CoderHam) [reply src/backends/onnx/onnx_backend.cc:862]: If you notice the string_cuda_copy is never used after initializing. It is always false. Thus this is never run

Comment (GuanLuo) [reply src/backends/onnx/onnx_backend.cc:862]: Keep `string_cuda_copy` and change line 840 to `string_cuda_copy |= cuda_used;`. FYI, @tanmayv25 this should fix the L0_infer-cudashm

File: src/core/cuda_utils.h (hunk 2 above)
Comment (CoderHam) [reply src/core/cuda_utils.h]: You're right. Removed.

File: src/backends/onnx/onnx_backend.cc (hunk 1 above)
Comment (CoderHam) [reply src/backends/onnx/onnx_backend.cc]: Sure. Made this fix.

Comment (tanmayv25) [reply src/backends/onnx/onnx_backend.cc]: @GuanLuo , I did try that yesterday evening but the backend  still read weird values from the memory. Did you check this change fixed L0_infer_cudashm?

Comment (GuanLuo) [reply src/backends/onnx/onnx_backend.cc]: I didn't check, just noticed that in this PR. It's weird that the issue is not due to synchronization as it only fails on cudashm variation. I will go over this part more closely once I get to it.

**Pull request merged**